std/socket.d

```diff
@@ -3139,17 +3139,21 @@ public:
         version (Windows)
         {
             auto read = .recvfrom(sock, buf.ptr, capToInt(buf.length), cast(int) flags, from.name, &nameLen);
-            from.setNameLen(nameLen);
-            assert(from.addressFamily == _family);
-            // if (!read) //connection closed
+            if (read >= 0)
+            {
+                from.setNameLen(nameLen);
+                assert(from.addressFamily == _family);
+            }
             return read;
         }
         else
         {
             auto read = .recvfrom(sock, buf.ptr, buf.length, cast(int) flags, from.name, &nameLen);
-            from.setNameLen(nameLen);
-            assert(from.addressFamily == _family);
-            // if (!read) //connection closed
+            if (read >= 0)
+            {
+                from.setNameLen(nameLen);
+                assert(from.addressFamily == _family);
+            }
             return read;
         }
```

Comment:
After the `recvfrom` call, both branches are identical, so there's an opportunity to DRY